R/legend-utils.R

```diff
@@ -0,0 +1,135 @@
+#' @title SetLegendPosition
+#' @param plotHandle Graphical object created from ggplot
+#' @param plotConfiguration plot Configuration for legend
+#' @param dataMapping mapping (might be overwritten to use plotConfiguration instead)
+#' @return Current plot with Legend Layer
+#' @description
+#' SetLegend set a Legend Position and labels in a plot
+#' @export
+setLegend <- function(plotHandle, plotConfiguration, dataMapping) {
+
+  # Check Inputs
+  stopifnot(class(plotHandle) %in% c("gg", "ggplot"))
+
+  # Set Legend Position according to plotConfiguration
+  # plotConfiguration will be added when implemented
+  plotHandle <- setLegendPosition(plotHandle)
+
+  # Redefine label of groups in legend
+  if (is.null(dataMapping$colorGrouping)) {
+    plotHandle <- plotHandle + guides(color = "none")
+  } else {
+    plotHandle <- plotHandle + guides(color = guide_legend(data$colorGrouping))
+  }
+  if (is.null(dataMapping$sizeGrouping)) {
+    plotHandle <- plotHandle + guides(size = "none")
+  } else {
+    plotHandle <- plotHandle + guides(size = guide_legend(data$sizeGrouping))
+  }
+  if (is.null(dataMapping$shapeGrouping)) {
+    plotHandle <- plotHandle + guides(shape = "none")
+  } else {
+    plotHandle <- plotHandle + guides(shape = guide_legend(data$shapeGrouping))
+  }
+
+  return(plotHandle)
+}
+
+#' @title SetLegendPosition
+#' @param plotHandle Graphical object created from ggplot
+#' @param Location Inside/Oustide/None
+#' @param X.Location NA/Left/Center/Right
+#' @param Y.Location NA/Top/Center/Bottom
+#' @param Subplot.Index (optional) OPTION NOT IMPLEMENTED YET
+#' @return Current plot with Legend Layer
+#' @description
+#' SetLegendPosition set a Legend Position in a plot (case insensitive)
+#' Default legend setting is within plot on the top right corner
+#' @export
+setLegendPosition <- function(plotHandle,

```

Comment:
@pchelle We have two methods `setLegentPosition` 
One in `legend-util`, and one in `utilities-legend`. Time to clean that up please. Going to create an issue for this

Reply:
https://github.com/Open-Systems-Pharmacology/TLF-Library/issues/76